File: backend/Ecommarce.Api/Models/AdminDashboardModels.cs
namespace Ecommarce.Api.Models;

public record DashboardStats(
  decimal TotalRevenue,
  string TotalOrders,
  string DeliveredOrders,
  string OrdersLeft,
  string ReturnedOrders,
  string CustomerQueries,
  decimal TotalPurchaseCost,
  decimal AverageSellingPrice,
  decimal ReturnValue,
  string ReturnRate
);

public record OrderItem(
  string Id,
  string CustomerName,
  string Date,
  decimal Amount,
  string Status
);

public record PopularProduct(
  string Id,
  string Name,
  string SoldCount,
  decimal Price,
  string ImageUrl
);

File: frontend/src/app/admin/models/admin-dashboard.models.ts
export interface DashboardStats {
  totalRevenue: number;
  totalOrders: string;
  deliveredOrders: string;
  ordersLeft: string;
  returnedOrders: string;
  customerQueries: string;
}

export interface OrderItem {
  id: string;
  customerName: string;
  date: string;
  amount: number;
  status: 'Completed' | 'Pending' | 'Shipped';
}

export interface PopularProduct {
  id: string;
  name: string;
  soldCount: string;
  price: number;
  imageUrl: string;
}
